sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java

```diff
@@ -28,8 +28,18 @@
  */
 @Experimental
 public interface BeamSqlSeekableTable extends Serializable{
+  /**
+   * prepare the instance.
+   */
+  void setup();
+
   /**
    * return a list of {@code Row} with given key set.
    */
   List<Row> seekRow(Row lookupSubRow);
+
+  /**
+   * cleanup resources of the instance.
+   */
+  void teardown();
```

Comment:
`setUp` & `tearDown` is optional, so maybe we can declare them as default methods?

```java
default void tearDown() {
}
```

Reply:
make sense as it's optional.